packages/readmodel/src/utils.ts

```diff
@@ -18,3 +23,29 @@ export const makeDrizzleConnection = (
   });
   return drizzle({ client: pool });
 };
+
+export const makeUniqueKey = (ids: string[]): string => ids.join("#");
+
+export const checkMetadataVersion = async <
+  T extends AnyPgTable & {
+    metadataVersion: AnyPgColumn<{ data: number }>;
+    id: AnyPgColumn;
+  }
+>(
+  tx: DrizzleTransactionType,
+  table: T,
+  metadataVersion: number,
+  id: string,
+  filter: SQL<unknown> = eq(table.id, id)
+): Promise<boolean> => {
+  const existingMetadataVersion = (
+    await tx
+      .select({
+        metadataVersion: table.metadataVersion,
+      })
+      .from(table as AnyPgTable)
+      .where(filter)
+  )[0]?.metadataVersion;
+
+  return !existingMetadataVersion || existingMetadataVersion <= metadataVersion;
```

Comment:
This is just a small suggestion for readability, but I don't have a strong opinion! Could you also add some JSDoc to this function to explain how to use it, what it does, and what the parameters are?

Thanks 🙏 

```suggestion
  const [row] = await tx
    .select({
      existingMetadataVersion: table.metadataVersion,
    })
    .from(table)
    .where(filter);

  const existingMetadataVersion = row?.existingMetadataVersion;

  // Return `true` if there's no row or no metadataVersion in the DB,
  // or if the existing version <= the new version.
  if (existingMetadataVersion == null) {
    return true;
  }

  return existingMetadataVersion <= newMetadataVersion;
```

Reply:
We have integrated your proposal (the jsdoc too) . Once the this PR is merged into `develop`, we will update (one at a time) the others that will call the `checkMetadataVersion`.